### DISCUSSION TITLE
Getting values at time 'x' from layer 'y' based on phenometrics derived from layer 'z'

### DISCUSSION THREAD

**Discussion by Antguz** [davidfrantz/force#325] [Q&A]
Dear community,

I am writing to seek guidance to approach a method. I am looking to derive layers for a interpolation on an specific band based on dates of land surface phenology derived from another an index or band. For instance, I would like to get values of the 16-day interpolated blue band that match with the dates of the start and end of the season of polar metrics derived from kNDVI. 

So far my approach is to derive both, the interpolation and the LSP, and then extract pixels according LSP dates. However, this implies a significantly increasing storage (especially for the 16-day interpolations). Is there any way to do such computation on-the fly?

Thanks in advance for your help!

AG.

**Comment by davidfrantz** [comment]
Dear AG,

what you describe is very similar to what we termed *phenology-adaptive  compositing (PAC)*. Please have a look at this article:
https://doi.org/10.1016/j.rse.2017.01.002

This is implemented in FORCE in the Level 3 submodule. However, it does not include the option for interpolation... This step would need to be done first - like you do it right now.

If you already have code, writing a UDF for the TSA submodule might be a way of doing the interpolation on the fly. However, there is currently no way of integrating a co-layer like LSP...

That said, the functionality in FORCE is pretty much there. However, the specific combination and need for on-the-fly is currently not available...

One alternative solution to overcome the storage limitations, would be to use fairly small tiles, then use option 1 (PAC) on a tile-by-tile basis with deleting intermediate steps in each iteration (you would need to write a small wrapper - or use the ``force-magic-parameters`` functionality).

I hope this somewhat helps,
David

**Reply by Antguz** [reply]
Thank you David, your guide is always helpful!
